Automattic/jetpack
# Protect: Add scan progress bar

#### Description
This PR adds a progress bar to the in-progress scan for the current scan.

#### Changes proposed in this Pull Request:

* Adds a `currentProgress` property to the `status` object when a current scan is accessible
* Adds a custom `ProgressBar` component
* Makes use of the `ProgressBar` component on the admin page for the in-progress display using the `value` status property

#### Other information:

#### Jetpack product discussion


- 1201069996155224-as-1203251534024903

#### Does this pull request change what data or activity we track or use?


- No

#### Testing instructions:





* Fire up a Jurassic Ninja instance w/ the Jetpack Beta plugin installed using this Protect branch
* Upgrade to Scan inclusive plan
* Install a known threat
* Queue a scan from the UI and verify that the in-progress screen displays a progress bar that matches the status of the scan (verified referring to the progress in RWDB)
* Ensure all related actions remain functional and perform as expected

#### Screenshots:
<img width="1329" alt="Screen Shot 2022-10-28 at 06 47 13" src="https://user-images.githubusercontent.com/43220201/198748085-b4feeb82-fd96-4b85-ab08-66962d79adde.png">
